src/commands/test/test_collector.py

```diff
@@ -1,93 +1,128 @@
 import os
```

Comment:
I think collector refactor and related tests could definitely be a separate PR

Reply:
Maybe if it's not a problem you can split it into two, would make it much easier to review...

Reply:
I'll try. I should've used bottom-up approach to refactoring.